lib/screens/HomeScreen.tsx

```diff
@@ -589,7 +589,15 @@ const HomeScreen: React.FC<HomeScreenProps> = ({ navigation, route }) => {
           >
             <Text style={styles.pfpText}>{userInitials}</Text>
           </TouchableOpacity>
-          <Image source={require('../../assets/icon.png')} style={{width: width * 0.105, height: width * 0.105, marginEnd: width * 0.435}} />
+          <Image source={require('../../assets/icon.png')} style={{width: width * 0.105, height: width * 0.105, marginEnd: width * 0.025}} />
+          <TouchableOpacity onPress={() => setIsSearchVisible(!isSearchVisible)}>
+            <Ionicons
+              testID="searchButton"
+              name={isSearchVisible ? "close-outline" : "search-outline"} // Switch between "X" and search icon
+              size={36}
+              color={theme.text}
+            />
```

Comment:
So for name isSearchVisible we check for true and false and determine the outline. Well done.